python/pyarrow/types.py

```diff
@@ -407,155 +223,71 @@ def is_unicode(t):
     return is_string(t)
 
 
+@doc(is_null, datatype="string (utf8 unicode)")
 def is_string(t):
-    """
-    Return True if value is an instance of string (utf8 unicode) type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id == lib.Type_STRING
 
 
+@doc(is_unicode, method="is_large_string")
 def is_large_unicode(t):
-    """
-    Alias for is_large_string.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return is_large_string(t)
 
 
+@doc(is_null, datatype="large string (utf8 unicode)")
 def is_large_string(t):
-    """
-    Return True if value is an instance of large string (utf8 unicode) type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id == lib.Type_LARGE_STRING
 
 
+@doc(is_null, datatype="fixed size binary")
 def is_fixed_size_binary(t):
-    """
-    Return True if value is an instance of a fixed size binary type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id == lib.Type_FIXED_SIZE_BINARY
 
 
+@doc(is_null, datatype="date")
 def is_date(t):
-    """
-    Return True if value is an instance of a date type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id in _DATE_TYPES
 
 
+@doc(is_null, datatype="date32 (days)")
 def is_date32(t):
-    """
-    Return True if value is an instance of a date32 (days) type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id == lib.Type_DATE32
 
 
+@doc(is_null, datatype="date64 (milliseconds)")
 def is_date64(t):
-    """
-    Return True if value is an instance of a date64 (milliseconds) type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id == lib.Type_DATE64
 
 
+@doc(is_null, datatype="logical map")
```

Comment:
Do we want to keep "logical" in `logical map`?

Reply:
The underlying implementation is `@GARROW_TYPE_MAP: A repeated struct logical type.`

Reply:
I think we can remove that. We don't have a clear definition of "logical" types in the arrow format spec, and the fact that a map type is basically a list of structs is an implementation detail (and adding "logical" here doesn't help making that clear)